### PR TITLE
[User] Review food

### PR DESCRIPTION
**Pull request by minhld99**
## Related Tickets
- [#39965](https://edu-redmine.sun-asterisk.vn/issues/39965)

## WHAT
- Review food with star rating and text
- Reply to review without star rating
- Add animate to newly added comments

## HOW
- N/A

## WHY
- N/A

## Evidence (Screenshot or Video)
food_review

### CODE REVIEW

**Comment by minhld99** [comment]
ready

File: main/templates/foods/details.html [hunk 1]
```diff
@@ -12,21 +12,22 @@
...
 {% block css %}
     <link rel="stylesheet" href="{% static 'css/bootstrap.min.css' %}">
+    <link rel="stylesheet" href="{% static 'css/animate.min.css' %}">
 {% endblock %}
 
 {% block content %}
     <!-- breadcrumb -->
-    <div style="margin-top: 110px;"></div>
-    <ol class="breadcrumb" style="justify-content: center; text-align: center;">
+    <div class="breadcrumbMargin"></div>
```
Comment (huongnt-2545) [inline main/templates/foods/details.html:20]: class, id nên đặt theo style: `breadcrumb-margin`, `detail-wrap`, ..

File: main/urls.py [hunk 2]
```diff
@@ -13,4 +13,6 @@
     path('password-reset-complete/', auth_views.PasswordResetCompleteView.as_view(template_name='accounts/password_reset_complete.html'), name='password_reset_complete'),
     path('search/', views.index, name='search'),
     path('food/<int:id>/details/', views.food_details, name='food-details'),
+    path('food/<int:id>/details/review/', views.review, name='review'),
+    path('food/<int:id1>/details/review/<int:id2>/reply/', views.reply, name='reply'),
```
Comment (huongnt-2545) [inline main/urls.py:17]: id1 => food_id
id2 => review_id

File: main/views.py [hunk 3]
```diff
@@ -54,3 +60,42 @@ def food_details(request, id):
         "rate_dict": _rate,
     }
     return render(request, 'foods/details.html', context)
+
+@login_required
+def review(request, id):
+    if request.method == 'POST':
+        user = request.user
+        food = Food.objects.get(id=id)
```
Comment (huongnt-2545) [inline main/views.py:68]: check trường hợp object not exist
nên dùng filter() rồi lấy first()

File: main/views.py [hunk 4]
```diff
@@ -54,3 +60,42 @@ def food_details(request, id):
...
+            "review_id": review_id,
+        }
+
+        return JsonResponse(context)
+        
+@login_required
+def reply(request, id1, id2):
+    if request.method == 'POST':
+        user = request.user
+        parent = Review.objects.get(id=id2)
```
Comment (huongnt-2545) [inline main/views.py:88]: nên dùng filter() rồi lấy first() để ko raise exception

**Comment by minhld99** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**